### PR TITLE
Forward compatibility include `<Kokkos_ReductionIdentity.hpp>` header

### PR DESCRIPTION
**Pull request by dalg24**
... wherever appropriate

This is a draft.  We will need it if kokkos/kokkos#5450 gets merged

### CODE REVIEW

File: src/geometry/ArborX_Box.hpp [hunk 1]
```diff
@@ -18,6 +18,9 @@
 #include <ArborX_Point.hpp>
 
 #include <Kokkos_Macros.hpp>
+#if __has_include(<Kokkos_ReductionIdentity.hpp>) // since Kokkos 4.0
```
Comment (aprokop) [inline src/geometry/ArborX_Box.hpp:21]:
```suggestion
#if __has_include(<Kokkos_ReductionIdentity.hpp>) // FIXME remove once require Kokkos 4.0
```

Comment (dalg24) [reply src/geometry/ArborX_Box.hpp:21]: How strong do you feel about this because I think my comment was good enough and I don't like the suggested change

**Review by masterleinad — APPROVED** [review]

File: src/geometry/ArborX_Box.hpp (hunk 1 above)
Comment (aprokop) [reply src/geometry/ArborX_Box.hpp:21]: I'm fine either way. I just wanted to have a simple way to stumble upon it in the future, so feel like adding a FIXME would be nice.

Comment (dalg24) [reply src/geometry/ArborX_Box.hpp]: Done

**Review by aprokop — APPROVED** [review]

**Pull request merged**